lib.rs

```diff
@@ -350,14 +419,25 @@ impl<A: Array> SmallVec<A> {
     /// ```
     #[inline]
     pub fn from_vec(mut vec: Vec<A::Item>) -> SmallVec<A> {
-        let (ptr, cap, len) = (vec.as_mut_ptr(), vec.capacity(), vec.len());
-        mem::forget(vec);
+        if vec.capacity() <= A::size() {
```

Comment:
The documentation says that `from_vec` does not copy the elements.  I think we should keep the current behavior of keeping the items on the heap.  Users who want to move them inline if possible can use `shrink_to_fit` or other methods.

Reply:
This is unfortunately not possible with the new design: there is no way of representing a heap-allocated buffer with a capacity less than `A::size()`. If `self.capacity <= A::size()` then the code assumes that the data is in the inline buffer.

Reply:
Oh, right.  In that case the documentation should be updated.